index.js

```diff
@@ -1,257 +1,245 @@
+        });
+});
+
+
+// get movie by title
+app.get('/movies/:Title',(req,res)=>{
+    Movies.findOne({Title: req.params.Title})
+        .then((movie) => {
+            res.status(201).json(movie);
+        })
+        .catch((error) => {
+            console.error(error);
+            res.status(500).send('Error ' + error);
+        })  
+});
+
+// get director by name
+app.get('/directors/:Name', (req,res) => {
+    Directors.findOne({Name: req.params.Name})
+        .then((director) => {
+            res.status(201).json(director);
+        })
+        .catch((error) => {
+            console.error(error);
+            res.status(500).send('Error: ' + error);
+        })
+})  
+
+// get genre by name
+app.get('/genres/:Name',(req,res)=>{
+    Genres.findOne({Name: req.params.Name})
+        .then((genre) => {
+            res.status(201).json(genre);
+        })
+        .catch((error)=>{
+            console.error(error);
+            res.status(500).send('Error: ' + error);
+        })
 })
 
 // get user by id
 app.get('/users/:id',(req,res)=>{
     res.json(users.find((user)=>{
         return req.params.id === user.id}));
-    })
+    });
 
-// add user 
+// add user
 app.post('/users',(req,res)=>{
-    let newUser = req.body;
-
-    if (!newUser.username){
-        const message = 'Username is missing in request body';
-        res.status(400).send(message);
-    } else {  
-        newUser.id = uuid.v4();
-        users.push(newUser);
-        res.status(201).send('User added');
-}
+    Users.findOne({Username:req.body.Username})
+        .then((user)=>{
+            if(user){
+                return res.status(400).send(req.body.Username + 'already exists');
+            } else {
+                Users.create({
+                    Username: req.body.Username,
+                    Password: req.body.Password,
+                    Email:req.body.Email,
+                    Birthday: req.body.Birthday
+                })
+                .then((user)=>{res.status(201).json(user)})
+                .catch((error)=>{
+                    console.error(error);
+                    res.status(500).send('Error: ' + error)
+                })
+            }
+        })
+        .catch((error)=>{
+            console.error(error);
+            res.status(500).send('Error: ' + error);
+        })
+});
+
+// update user data
+app.post('/users/:Username',(req,res)=>{
+    Users.findOneAndUpdate({Username: req.params.Username},
+        {$set:
+            {
+                Username: req.body.Username,
+                Password: req.body.Password,
+                Email:req.body.Email,
+                Birthday: req.body.Birthday
+            }
+        },
+        {new: true}) // this makes sure that the updated document is gonna returned
+        .then((updatedUser) => {
+            res.status(201).json(updatedUser)})
+        .catch((error) => {
+            console.error(error);
+            res.status(500).send('Error: ' + error)})
```

Comment:
This looks like an update endpoint yet I see a post method is being used. Please help cross check on the best [Http Verb](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) to use.